### PR TITLE
Show I18n value in header when available

### PR DESCRIPTION
**Pull request by tboyko**
In response to #97

### CODE REVIEW

**Comment by eitoball** [comment]
Thank you for PR. Could you do the following?

 1. Add specs.
 1. Squash commits into one.
 1. Rebase to current master.

Thank you!

File: lib/comma/extractors.rb [hunk 1]
```diff
@@ -19,24 +19,30 @@ def id(*args)
...
     def method_missing(sym, *args, &block)
-      @results << sym.to_s.humanize if args.blank?
+
+      @results << header_string(sym) if args.blank?
       args.each do |arg|
         case arg
         when Hash
           arg.each do |k, v|
-            @results << ((v.is_a? String) ? v : v.to_s.humanize)
+            @results << ((v.is_a? String) ? v : header_string(sym))
```
Comment (eitoball) [inline lib/comma/extractors.rb]: Should be `header_string(v)`?

File: lib/comma/extractors.rb [hunk 2]
```diff
@@ -19,24 +19,30 @@ def id(*args)
...
           @results << arg
         else
           raise "Unknown header symbol #{arg.inspect}"
         end
       end
     end
+
+    def header_string attribute
+      key = "activerecord.attributes.#{@instance.class.to_s.underscore}.#{attribute.to_s}"
+      I18n.t(key, default: attribute.to_s.humanize)
```
Comment (eitoball) [inline lib/comma/extractors.rb]: When translation for `key` is missing, will it use value in `default` option without raising error?

Comment (alanq) [inline lib/comma/extractors.rb]: Would this work for header_string() ?
It gives a humanized default if there's no translation key or no such attribute
`@instance.class.human_attribute_name attribute`

**Comment by tboyko** [comment]
@alanq That seems like a better method. Thanks!

@eitoball Correct on the question about using `default`

Bigger problem at the moment is that this code changes the default behavior of the gem. Perhaps @eitoball can suggest how we might add a configuration option to enable this functionality globally for the gem? Or, perhaps the major version of the gem is bumped up?

**Comment by eitoball** [comment]
> Bigger problem at the moment is that this code changes the default behavior of the gem. Perhaps @eitoball can suggest how we might add a configuration option to enable this functionality globally for the gem? Or, perhaps the major version of the gem is bumped up?

Can you change so that it will not change the default behavior? If you need some way to configure, let's discuss later.

**Comment by tboyko** [comment]
For future readers, here's an easy way to get this functionality in comma 4.X:

```ruby
Comma::HeaderExtractor.value_humanizer = lambda do |value, _model_class|
  value.is_a?(String) ? value : _model_class.human_attribute_name(value)
end
```

You can put this in the initializers folder if you're using Rails.

**Pull request closed without merging**